packages/cli/src/zosjobs/submit/Submit.shared.handler.ts

```diff
@@ -31,150 +31,164 @@ export default class SharedSubmitHandler extends ZosmfBaseHandler {
+                    if (parms.volume) {
+                        options.volume = parms.volume;
+
+                        // Get JCL from data set or member
+                        const getJcl = await Get.dataSet(this.mSession, this.mArguments.dataset, options);
+                        source = this.mArguments.dataset;
+
+                        apiObj = await SubmitJobs.submitJclString(this.mSession, getJcl.toString(), parms);
+                        if (parms.viewAllSpoolContent) {
+                            spoolFilesResponse = apiObj;
+                        }
+
+                        break;
+                    } else {
+                        response = await SubmitJobs.submitJobCommon(this.mSession, {jobDataSet: this.mArguments.dataset,
+                            jclSymbols: this.mArguments.jclSymbols});
+                        apiObj = await SubmitJobs.checkSubmitOptions(this.mSession, parms, response);
+                        source = this.mArguments.dataset;
+
+                        if (parms.viewAllSpoolContent) {
+                            spoolFilesResponse = apiObj;
+                        }
                     }
 
                     break;
-                } else {
-                    response = await SubmitJobs.submitJobCommon(this.mSession, {jobDataSet: this.mArguments.dataset,
+                // Submit JCL from a USS file
+                case "uss-file":
+                    response = await SubmitJobs.submitJobCommon(this.mSession, {jobUSSFile: this.mArguments.file,
                         jclSymbols: this.mArguments.jclSymbols});
                     apiObj = await SubmitJobs.checkSubmitOptions(this.mSession, parms, response);
-                    source = this.mArguments.dataset;
+                    source = this.mArguments.ussfile;
 
                     if (parms.viewAllSpoolContent) {
                         spoolFilesResponse = apiObj;
                     }
-                }
 
-                break;
-            // Submit JCL from a USS file
-            case "uss-file":
-                response = await SubmitJobs.submitJobCommon(this.mSession, {jobUSSFile: this.mArguments.file,
-                    jclSymbols: this.mArguments.jclSymbols});
-                apiObj = await SubmitJobs.checkSubmitOptions(this.mSession, parms, response);
-                source = this.mArguments.ussfile;
-
-                if (parms.viewAllSpoolContent) {
-                    spoolFilesResponse = apiObj;
-                }
+                    break;
+                // Submit the JCL from a local file
+                case "local-file": {
+                    parms.jclSource = this.mArguments.localFile;
+                    try {
+                        const JclString = fs.readFileSync(this.mArguments.localFile).toString();
+                        apiObj = await SubmitJobs.submitJclString(this.mSession, JclString, parms);
```

Comment:
Please move this line and the following one outside of (below) the `try-catch` block. We only want to show an FS error if the `fs.readFileSync` call fails. REST API errors thrown by the `SubmitJobs` API can be handled by the outer try-catch block.